nyx_client/nyx_client/client.py

```diff
@@ -593,26 +606,35 @@ def create_data(
             "name": name,
             "title": title,
             "description": description,
-            "size": size,
             "genre": genre,
             "categories": categories,
             "lang": lang,
             "status": status,
             "preview": preview_base64_string,
-            "downloadURL": download_url,
             "contentType": content_type,
         }
+        if download_url:
+            data["download_url"] = download_url
+            data["size"] = (size,)
+            if not size:
+                raise ValueError("size is required when using download_url")
+        if file:
+            file.seek(0, os.SEEK_END)
```

Comment:
(As discussed) It'd be nice for the server to do this so we don't have to seek in the input stream. Let's see what we decide tomorrow.

Reply:
Removed this now.